config/source/source.go

```diff
@@ -0,0 +1,40 @@
+// Package source is a base for all config sources.
+package source
+
+import (
+	"fmt"
+	"net/url"
+
+	"go-micro.dev/v5/codecs"
+)
+
+// Data holds a single config file marshaled to map[string]any.
+type Data struct {
+	// Source URL.
+	URL *url.URL
+	// Data holder.
+	Data map[string]any
+	// The Marshaler used to create Data -> map[string]any
+	Marshaler codecs.Marshaler
+	// If there was an error while processing the URL.
+	Error error
+
+	// AdditionalConfigs is a list of configs that we also have to read.
+	// or that have been injected by config.Read().
+	AdditionalConfigs []*url.URL
+}
+
+// Source is a config source.
+type Source interface {
+	fmt.Stringer
```

Comment:
Maybe make the string method here explicit, as we do in the other interfaces.

Reply:
Still not sure which one is nicer. Explicit on as a Method, explicit as an Interface which you depend on at many places.

Reply:
The benefit of explicit is that you can comment it and describe what it does